### PR TITLE
solving unweighted hons issue

### PR DESCRIPTION
**Pull request by VinsRR**
The higher-order networks were unweighted. This snippet:
```
dag_data = pp.DAGData(pp.IndexMap(list("01234")))
dag_data.append_walk(list("023"), weight=20)
dag_data.append_walk(list("124"), weight=30)

m = pp.MultiOrderModel.from_DAGs(dag_data, max_order=2)

hon_1 = m.layers[1]
hon_2 = m.layers[2]
print(hon_1.data.edge_weight)
print(hon_2.data.edge_weight)
```
Gives this:
```
tensor([1., 1., 1., 1.])
tensor([1., 1.])
```

### CODE REVIEW

File: src/pathpyG/core/MultiOrderModel.py [hunk 1]
```diff
@@ -248,7 +248,7 @@ def from_DAGs(
         dag_graph = next(iter(DataLoader(dag_data.dags, batch_size=len(dag_data.dags)))).to(config["torch"]["device"])
         edge_index = dag_graph.edge_index
         node_sequence = dag_graph.node_sequence
-        if dag_graph.edge_attr is None:
+        if dag_graph.edge_weight is None:
             edge_weight = torch.ones(edge_index.size(1), device=edge_index.device)
         else:
             edge_weight = dag_graph.edge_attr
```
Comment (M-Lampert) [inline src/pathpyG/core/MultiOrderModel.py:254]: Shouldn't this also be called `dag_graph.edge_weight` instead of `dag_graph.edge_attr`?

**Review by M-Lampert — COMMENTED** [review]
It would also be nice to have a unit-test that would check both cases, i.e. if an unweighted input returns only the counts of each topologically unique path and if a weighted input returns the aggregated weights. Maybe you can add that to `tests/core/test_multi_order_model.py`

File: src/pathpyG/core/MultiOrderModel.py (hunk 1 above)
Comment (VinsRR) [reply src/pathpyG/core/MultiOrderModel.py:254]: Yes, also that line needs to be modified. Thank you for noticing it!

**Comment by M-Lampert** [comment]
BTW: This is related to #155

**Comment by M-Lampert** [comment]
Could be closed due to #173, but best practice would be to merge it into #173 instead to have a clearer/cleaner merge history. @VinsRR can you either close it or merge it?

**Pull request merged**